Build failure: poetry on macOS ARM64

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *poetry* on **macOS ARM64** from current master (so that #281103 is fixed) with `nix run github:NixOS/nixpkgs#poetry`
2. It fails

### Build log

It boils down to: 

```
FAILED tests/console/commands/test_shell.py::test_shell - shellingham.posix.ps.PsNotAvailable: ps not found
```

[Full log](https://gist.github.com/Brawl345/d1ab7dad0a5d3f939cfec9eca394973e#file-poetry-build-failure-txt)

### Additional context

Building works on Linux ARM64. Don't know about macOS x64.

### Notify maintainers

@dotlambda ?

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.2.0, macOS 14.2.1`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/nix/store/xn8svimhlimx8mn19llhndm30cs01k0a-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/package-management/poetry/unwrapped.nix
+++ b/pkgs/tools/package-management/poetry/unwrapped.nix
@@ -37,6 +37,7 @@
 , pytest-mock
 , pytest-xdist
 , pythonAtLeast
+, darwin
 }:
 
 buildPythonPackage rec {
@@ -110,6 +111,8 @@ buildPythonPackage rec {
     httpretty
     pytest-mock
     pytest-xdist
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.ps
   ];
 
   preCheck = (''

